nautobot/circuits/views.py

```diff
@@ -4,277 +4,189 @@
-    model_form = forms.ProviderCSVForm
-    table = tables.ProviderTable
-
-
-class ProviderBulkEditView(generic.BulkEditView):
-    queryset = Provider.objects.annotate(count_circuits=count_related(Circuit, "provider"))
-    filterset = filters.ProviderFilterSet
-    table = tables.ProviderTable
-    form = forms.ProviderBulkEditForm
-
-
-class ProviderBulkDeleteView(generic.BulkDeleteView):
-    queryset = Provider.objects.annotate(count_circuits=count_related(Circuit, "provider"))
-    filterset = filters.ProviderFilterSet
-    table = tables.ProviderTable
-
-
-#
-# ProviderNetwork
-#
-
-
-class ProviderNetworkListView(generic.ObjectListView):
-    queryset = ProviderNetwork.objects.all()
-    filterset = filters.ProviderNetworkFilterSet
-    filterset_form = forms.ProviderNetworkFilterForm
-    table = tables.ProviderNetworkTable
-
-
-class ProviderNetworkView(generic.ObjectView):
-    queryset = ProviderNetwork.objects.all()
-
-    def get_extra_context(self, request, instance):
-        circuits = (
-            Circuit.objects.restrict(request.user, "view")
-            .filter(Q(termination_a__provider_network=instance.pk) | Q(termination_z__provider_network=instance.pk))
-            .prefetch_related("type", "tenant", "terminations__site")
-        )
-
-        circuits_table = tables.CircuitTable(circuits)
-        circuits_table.columns.hide("termination_a")
-        circuits_table.columns.hide("termination_z")
-
-        paginate = {"paginator_class": EnhancedPaginator, "per_page": get_paginate_count(request)}
-        RequestConfig(request, paginate).configure(circuits_table)
-
-        return {
-            "circuits_table": circuits_table,
-        }
-
-
-class ProviderNetworkEditView(generic.ObjectEditView):
-    queryset = ProviderNetwork.objects.all()
-    model_form = forms.ProviderNetworkForm
-
-
-class ProviderNetworkDeleteView(generic.ObjectDeleteView):
-    queryset = ProviderNetwork.objects.all()
-
-
-class ProviderNetworkBulkImportView(generic.BulkImportView):
-    queryset = ProviderNetwork.objects.all()
-    model_form = forms.ProviderNetworkCSVForm
-    table = tables.ProviderNetworkTable
-
-
-class ProviderNetworkBulkEditView(generic.BulkEditView):
-    queryset = ProviderNetwork.objects.all()
-    filterset = filters.ProviderNetworkFilterSet
-    table = tables.ProviderNetworkTable
-    form = forms.ProviderNetworkBulkEditForm
-
-
-class ProviderNetworkBulkDeleteView(generic.BulkDeleteView):
-    queryset = ProviderNetwork.objects.all()
-    filterset = filters.ProviderNetworkFilterSet
-    table = tables.ProviderNetworkTable
-
-
-#
-# Circuit Types
-#
-
-
-class CircuitTypeListView(generic.ObjectListView):
+class CircuitTypeUIViewSet(
+    view_mixins.ObjectDetailViewMixin,
+    view_mixins.ObjectListViewMixin,
+    view_mixins.ObjectEditViewMixin,
+    view_mixins.ObjectDestroyViewMixin,
+    view_mixins.ObjectBulkDestroyViewMixin,
+    view_mixins.ObjectBulkCreateViewMixin,
+):
```

Comment:
CircuitType doesn't implement a BulkUpdate view at present which is OK. This shows how powerful the `NautobotUIViewSetRouter` is. Since it doesn't have the `bulk_update` method defined, it nullifies the route built-in.